src/Infrastructure/Persistence/Specification/Extensions/ParameterReplacerVisitor.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Linq.Expressions;
+
+namespace FSH.WebApi.Infrastructure.Persistence.Specification.Extensions;
+
+internal sealed class ParameterReplacerVisitor : ExpressionVisitor
+{
+    private readonly Expression _newExpression;
+    private readonly ParameterExpression _oldParameter;
+
+    private ParameterReplacerVisitor(ParameterExpression oldParameter, Expression newExpression)
+    {
+        _oldParameter = oldParameter;
+        _newExpression = newExpression;
+    }
+
+    internal static Expression Replace(Expression expression, ParameterExpression oldParameter, Expression newExpression)
+    {
+        return new ParameterReplacerVisitor(oldParameter, newExpression).Visit(expression);
+    }
+
+    protected override Expression VisitParameter(ParameterExpression p)
+    {
+        if (p == _oldParameter)
+        {
+            return _newExpression;
+        }
+        else
+        {
+            return p;
+        }
+    }
+}
```

Comment:
I wonder if we could use EF Core's `ReplacingExpressionVisitor` in stead of this one. `ReplacingExpressionVisitor` is also used in https://github.com/fullstackhero/dotnet-webapi-boilerplate/blob/ae8fb818567bfe65d26931f5bed94d0d3f12e6ec/src/Infrastructure/Persistence/Context/AppendGlobalQueryFilterExtension.cs#L19

Reply:
I tried using EF Core's ReplacingExpressionVisitor without success.
The EF Core's one is more complicated https://github.com/dotnet/efcore/blob/main/src/EFCore/Query/ReplacingExpressionVisitor.cs
If we would like to use it would require refactoring this expression call https://github.com/fullstackhero/dotnet-webapi-boilerplate/blob/78cb89de033f6f9e86fa1beeace2d1d31fcecc0c/src/Infrastructure/Persistence/Specification/Extensions/NpgsqlSearchExtension.cs#L43

Unfortunately im not able to do that.

Reply:
Ok, nevermind then... was just wondering ;-)